libmamba/src/core/shell_init.cpp

```diff
@@ -569,15 +569,34 @@ namespace mamba
     )
     {
         auto out = Console::stream();
-        fmt::print(
-            out,
-            "Modifying RC file {}\n"
-            "Generating config for root prefix {}\n"
-            "Setting mamba executable to: {}\n",
-            fmt::streamed(file_path),
-            fmt::styled(fmt::streamed(conda_prefix), fmt::emphasis::bold),
-            fmt::styled(fmt::streamed(mamba_exe), fmt::emphasis::bold)
-        );
+
+        if (context.dry_run)
+        {
+            fmt::print("Running `shell init` in dry-run mode\n");
+            fmt::print("Running `shell init` would:\n");
+            fmt::print(
+                out,
+                " - would modify RC file: {}\n"
+                " - would generate config for root prefix: {}\n"
+                " - would set mamba executable to: {}\n",
+                fmt::streamed(file_path),
+                fmt::styled(fmt::streamed(conda_prefix), fmt::emphasis::bold),
+                fmt::styled(fmt::streamed(mamba_exe), fmt::emphasis::bold)
+            );
+        }
+        else
+        {
+            fmt::print("Running `shell init`, which:\n");
+            fmt::print(
+                out,
+                " - modifies RC file: {}\n"
+                " - generates config for root prefix: {}\n"
+                " - sets mamba executable to: {}\n",
+                fmt::streamed(file_path),
+                fmt::styled(fmt::streamed(conda_prefix), fmt::emphasis::bold),
+                fmt::styled(fmt::streamed(mamba_exe), fmt::emphasis::bold)
+            );
```

Comment:
The last fmt::print can be put outside the branching to avoid code duplication.

Reply:
The last `fmt::print` statement, or the last `fmt::styled` one? The `fmt::print` statements here are different.

Reply:
ah my bad, I read too fast. Please ignore this comment.